build_tools/cmake/iree_c_module.cmake

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(CMakeParseArguments)
+
+# iree_c_module()
+#
+# Parameters:
+# NAME: Name of target (see Note).
+# SRC: MLIR source file to compile into a c module.
+# H_FILE_OUTPUT: The H header file to output.
+# TRANSLATE_TOOL: Translation tool to invoke (CMake target).
+# FLAGS: Flags to pass to the translation tool (list of strings).
```

Comment:
Can you document the defaults here?

Reply:
Done. Also added those to `iree_bytecode_module`.